src/main/java/co/cask/gcp/gcs/sink/GCSMultiBatchSink.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright © 2015-2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.gcp.gcs.sink;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Macro;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.batch.Output;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.data.schema.UnsupportedTypeException;
+import co.cask.cdap.api.dataset.lib.KeyValue;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.batch.BatchRuntimeContext;
+import co.cask.cdap.etl.api.batch.BatchSink;
+import co.cask.cdap.etl.api.batch.BatchSinkContext;
+import co.cask.gcp.common.FileSetUtil;
+import co.cask.gcp.common.GCPUtils;
+import co.cask.gcp.common.RecordFilterOutputFormat;
+import co.cask.hydrator.common.batch.JobUtils;
+import co.cask.hydrator.common.batch.sink.SinkOutputFormatProvider;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import org.apache.avro.mapreduce.AvroJob;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.lib.output.FileOutputFormat;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+
+/**
+ * {@link GCSMultiBatchSink} that stores the data of the latest run of an adapter in GCS.
+ */
+@Plugin(type = BatchSink.PLUGIN_TYPE)
+@Name("GCSMultiFiles")
+@Description("Writes records to one or more avro files in a directory on Google Cloud Storage.")
+public class GCSMultiBatchSink extends BatchSink<StructuredRecord, NullWritable, StructuredRecord> {
+  private static final String TABLE_PREFIX = "multisink.";
+
+  private final GCSMultiBatchSinkConfig config;
+
+  public GCSMultiBatchSink(GCSMultiBatchSinkConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void initialize(BatchRuntimeContext context) throws Exception {
+    super.initialize(context);
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    config.validate();
+    super.configurePipeline(pipelineConfigurer);
```

Comment:
there should be additional validation done. For example, if pipelineConfigurer.getStageConfigurer().getInputSchema() is not null, the plugin should check that the configured splitField exists in the input schema and is of type string or nullable string.

Reply:
There is not input schema during the configuration stage

Reply:
oh right because we expect various input schemas